src/helm-frontend/src/components/LeaderboardTables.tsx

```diff
@@ -32,10 +34,23 @@ export default function LeaderboardTables({
     ...groupsTables[activeGroup],
   });
   const [sortDirection, setSortDirection] = useState<number>(1);
+  const [filteredModels, setFilteredModels] =
+    useState<string[]>(modelsToFilter);
 
   useEffect(() => {
     setActiveGroupsTable({ ...groupsTables[activeGroup] });
-  }, [activeGroup, groupsTables]);
+    // upon receiving and setting data for current table, use sort to figure out n top models
+    if (nModelsToAutoFilter) {
+      const activeRows = groupsTables[0].rows;
+      const sortedRows = activeRows.sort((a, b) => {
+        return Number(b[1].value) - Number(a[1].value);
```

Comment:
Are the rows not already sorted?  In any case, write explicitly what your assumptions are.

Reply:
The rows are currently sorted by a column that is defined by a state variable (which is updated when the user clicks on a column). In this case we are always sorting by the same column (column 1) so it should occur separately in case the state variable changes. The state variable is not relevant in the case of the mini-leaderboard, but since  this component is used for both the mini-leaderboard and main leaderboard we need to maintain this separation. 

To resolve this, the leaderboard component does need a larger refactor to make the code cleaner, but functionality-wise everything works as is. 

Let me know if I should add a code comment for context.